dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumEventConsumer.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.CoreFeature
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.internal.domain.scope.RumApplicationScope
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.LongTaskEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+
+internal class RumEventConsumer(
+    private val dataWriter: DataWriter<Any>,
```

Comment:
Which writer is this? The one just putting data into file or the one making tracking consent consideration? If it's the latter, wouldn't it be cleaner to use specialised type here, for clarity?

Reply:
Is the one having the tracking consent also